### PR TITLE
Add validation for the date range of custom date range picker

### PR DESCRIPTION
**Pull request by sanjibroy360**
## Notion card
[https://www.notion.so/saeloun/Time-entry-report-Bug-96383e9d59df49a7bb625a60054f35ff](https://www.notion.so/saeloun/Time-entry-report-Bug-96383e9d59df49a7bb625a60054f35ff)

## Summary
- Ensure that in custom date range picker 'From' date input will be less than or equal to 'To' date and 'To' date input will be greater than or equal to 'From' date.

- When `From` date is greater than `To` date it will show this error message -
 `'From' date must be less than or equal to 'To' date`

- When `To` date is less than `From` date it will show this error message -
 `'To' date must be greater than or equal to 'From' date`

- For blank date field it will show this error message -
  `Date can not be blank`

- For invalid date it will show this error message -
  `Please enter a valid date`

- For now hid the share button from mobile and ipad view as well.

## Preview
[https://www.loom.com/share/437b7f08c5f6439b85f449e6a6040927](https://www.loom.com/share/437b7f08c5f6439b85f449e6a6040927)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking and retains same functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have annotated changes in the PR that are relevant to the reviewer
- [ ] I have added automated tests for my code

### CODE REVIEW

File: app/javascript/src/components/Reports/Header/index.tsx [hunk 1]
```diff
@@ -115,9 +114,9 @@ const Header = ({
                 <FilterIcon className="mr-4" color="#7C5DEE" size={16} />{" "}
                 Filters
               </li>
-              <li className="flex items-center p-2 text-miru-dark-purple-400">
+              {/* <li className="flex items-center p-2 text-miru-dark-purple-400">
```
Comment (onkar-saeloun) [inline app/javascript/src/components/Reports/Header/index.tsx:117]: can you remove commented code?

**Review by onkar-saeloun — APPROVED** [review]
LGTM small request

File: app/javascript/src/components/Reports/Header/index.tsx (hunk 1 above)
Comment (sanjibroy360) [reply app/javascript/src/components/Reports/Header/index.tsx:117]: @onkar-saeloun  actually we are temporarily hiding this `Share` button. Once the share feature gets implemented from backend we will revert this change.

Comment (onkar-saeloun) [reply app/javascript/src/components/Reports/Header/index.tsx:117]: Okay Thanks.

**Pull request merged**